Test failure with h5py 2.7.1

With the latest h5py 2.7.1, astropy test [fails](https://buildd.debian.org/status/fetch.php?pkg=python-astropy&arch=amd64&ver=2.0.1-2%2Bb1&stamp=1505135572&file=log) here:
```
___________________________ test_metadata_too_large ____________________________

tmpdir = local('/tmp/pytest-of-buildd/pytest-0/test_metadata_too_large0')

    @pytest.mark.skipif('not HAS_H5PY or not HAS_YAML')
    def test_metadata_too_large(tmpdir):
        test_file = str(tmpdir.join('test.hdf5'))
    
        t1 = Table()
        t1['a'] = Column(data=[1, 2, 3])
        t1.meta["meta"] = "0" * (2**16 + 1)
    
        with catch_warnings() as w:
            t1.write(test_file, path='the_table', serialize_meta=True, overwrite=True)
        assert len(w) == 1
>       assert str(w[0].message).startswith(
            "Attributes could not be written to the output HDF5 "
            "file: Unable to create attribute (Object header message is too large)")
E       AssertionError: assert False
E        +  where False = <built-in method startswith of str object at 0x7f9324948a30>('Attributes could not be written to the output HDF5 file: Unable to create attribute (Object header message is too large)')
E        +    where <built-in method startswith of str object at 0x7f9324948a30> = 'Attributes could not be written to the output HDF5 file: Unable to create attribute (object header message is too large)'.startswith
E        +      where 'Attributes could not be written to the output HDF5 file: Unable to create attribute (object header message is too large)' = str(UserWarning(u'Attributes could not be written to the output HDF5 file: Unable to create attribute (object header message is too large)',))
E        +        where UserWarning(u'Attributes could not be written to the output HDF5 file: Unable to create attribute (object header message is too large)',) = <warnings.WarningMessage object at 0x7f931bbb3b90>.message
```

Problem is that expects a capital **`O`** in the warning: `(Object header message is too large)`, while h5py now has a small one: `(object header message is too large)`.
Source file is [`astropy/io/misc/tests/test_hdf5.py`](https://github.com/astropy/astropy/blob/d0e03a8c9b8535865858e308662ce2a57c8cfa53/astropy/io/misc/tests/test_hdf5.py#L497)

--- a/astropy/io/misc/tests/test_hdf5.py
+++ b/astropy/io/misc/tests/test_hdf5.py
@@ -492,6 +492,7 @@ def test_metadata_very_large_fails_compatibility_mode(tmpdir):
                  overwrite=True, compatibility_mode=True)
     assert len(w) == 2
 
+    # Error message slightly changed in h5py 2.7.1, thus the 2part assert
     assert str(w[1].message).startswith(
         "Attributes could not be written to the output HDF5 "
         "file: Unable to create attribute ")

--- a/astropy/io/misc/tests/test_hdf5.py
+++ b/astropy/io/misc/tests/test_hdf5.py
@@ -494,7 +494,8 @@ def test_metadata_very_large_fails_compatibility_mode(tmpdir):
 
     assert str(w[1].message).startswith(
         "Attributes could not be written to the output HDF5 "
-        "file: Unable to create attribute (Object header message is too large)")
+        "file: Unable to create attribute ")
+    assert "bject header message is too large" in str(w[1].message)
 
 
 @pytest.mark.skipif('not HAS_H5PY')

